### PR TITLE
[16.0] [MIG] ddmrp_report_part_flow_index

### PR DESCRIPTION
**Pull request by ThiagoMForgeFlow**
Migration of ddmrp_report_part_flow_index from 11.0 to 16.0

@ForegFlow

### CODE REVIEW

File: ddmrp_report_part_flow_index/security/ir.model.access.csv [hunk 1]
```diff
@@ -0,0 +1,3 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_ddmrp_flow_index_group,ddmrp.flow.index.group,model_ddmrp_flow_index_group,stock.group_stock_manager,1,1,1,1
+access_report_ddmrp_part_plan_flow_index,ddmrp.part.plan.flow.index,model_report_ddmrp_part_plan_flow_index,stock.group_stock_manager,1,1,1,1
```
Comment (LoisRForgeFlow) [inline ddmrp_report_part_flow_index/security/ir.model.access.csv]: Can you change this two access rights to the group ddmrp.group_ddmrp_manager and add a new one that allows readonly in index group by ddmrp.group_stock_buffer_maintainer

File: ddmrp_report_part_flow_index/views/ddmrp_flow_index_group_views.xml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+                            type="object"
+                            class="oe_stat_button"
+                            icon="fa-check"
+                        >
+                            <field
+                                name="active"
+                                widget="boolean_button"
+                                options='{"terminology": "active"}'
+                            />
+                        </button>
```
Comment (LoisRForgeFlow) [inline ddmrp_report_part_flow_index/views/ddmrp_flow_index_group_views.xml]: This needs migration

image

File: ddmrp_report_part_flow_index/views/stock_buffer_view.xml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+<odoo>
+
+    <record id="stock_buffer_view_form_inherited" model="ir.ui.view">
+        <field name="name">stock.buffer.form - Part Flow Index</field>
+        <field name="model">stock.buffer</field>
+        <field name="inherit_id" ref="ddmrp.stock_buffer_view_form" />
+        <field name="arch" type="xml">
+            <field name="qty_multiple" position="after">
+                <field name="flow_index_group_id" />
+            </field>
```
Comment (LoisRForgeFlow) [inline ddmrp_report_part_flow_index/views/stock_buffer_view.xml]: Can you move this to the header?

image

**Review by LoisRForgeFlow — COMMENTED** [review]
LG, some comments:

File: ddmrp_report_part_flow_index/views/ddmrp_flow_index_group_views.xml [hunk 4]
```diff
@@ -19,20 +19,6 @@
...
-                            type="object"
-                            class="oe_stat_button"
-                            icon="fa-check"
-                        >
-                            <field
-                                name="active"
-                                widget="boolean_button"
-                                options='{"terminology": "active"}'
-                            />
-                        </button>
```
Comment (LoisRForgeFlow) [inline ddmrp_report_part_flow_index/views/ddmrp_flow_index_group_views.xml]: Typically it is replaced by:

```
<widget
                        name="web_ribbon"
                        title="Archived"
                        bg_color="bg-danger"
                        attrs="{'invisible': [('active', '=', True)]}"
                    />
```

(not exaclty in the same position, you can check usages in other form views)

Comment (ThiagoMForgeFlow) [reply ddmrp_report_part_flow_index/views/ddmrp_flow_index_group_views.xml]: Thanks for the comments, I have applied the changes!

**Review by LoisRForgeFlow — APPROVED** [review]
All good now :+1: 

/ocabot merge nobump

**Pull request merged**